Reth 1.2.0

ZanCorDX opened the pull request:
## 📝 Summary

Lovely reth changes.

## 💡 Motivation and Context

Pectra

## ✅ I have completed the following steps:

* [X] Run `make lint`
* [X] Run `make test`
* [ ] Added tests (if applicable)

ZanCorDX committed:
starting

ZanCorDX committed:
running?

ryanschneider commented on crates/rbuilder/src/building/mod.rs:
```diff
@@ -626,18 +622,10 @@ impl<Tracer: SimulationTracer> PartialBlock<Tracer> {
-                    &ctx.initialized_cfg,
-                    &ctx.block_env,
-                )
+                .post_block_consolidation_requests_contract_call(db.as_mut(), &ctx.evm_env)
```
I'm pretty sure we need to call:

```
db.as_mut().merge_transitions(BundleRetention::Reverts);
```

somewhere in this block, e.g. before `Some(requests)` at line 641. h/t to Aresnii for spotting this on telegram!

ryanschneider replied:
With the refactor this is now called down on line 654 (since the db is no longer shadowed).

ferranbt committed:
Make changes

ferranbt committed:
More notes

ferranbt committed:
Revert again

ferranbt committed:
Final touch

ZanCorDX committed:
polishing

ZanCorDX committed:
lint

ZanCorDX committed:
block finalization fix

ferranbt committed:
Merge branch 'reth-1.2.0' into f-reth-1.2.0-op

dvush commented on crates/rbuilder/src/backtest/restore_landed_orders/resim_landed_block.rs:
```diff
@@ -10,20 +10,20 @@ use ahash::{HashMap, HashSet};
 impl ExecutedTxs {
     pub fn hash(&self) -> TxHash {
-        self.tx.hash()
+        *self.tx.hash()
```
I like this new change, lets return &TxHash instead here?

ryanschneider commented on crates/rbuilder/src/building/mod.rs:
```diff
@@ -137,7 +134,7 @@ impl BlockBuildingContext {
             )
             .ok()?;
-        block_env.coinbase = signer.address;
+        evm_env.block_env.coinbase = signer.address;
```
Do we need https://github.com/rkrasiuk/rbuilder/commit/d7b1637967dffc1f4f3ade5303aacc185963c5c3 /cc @rkrasiuk

ryanschneider reviewed (commented):
overall looks good to me, but curious if we need @rkrasiuk's bloblgas fix from his draft?

dvush reviewed (changes requested):
this branch breaks root hash and produces incorrect blocks a lot

no merge until fixed

avalonche committed:
Add revert protection and integration tests (#435)

## 📝 Summary

Adds revert protection and integration tests to the transactions in the
payload builder of the op-rbuilder.

## 💡 Motivation and Context


---

## ✅ I have completed the following steps:

* [x] Run `make lint`
* [x] Run `make test`
* [x] Added tests (if applicable)

SozinM committed:
New witness calculation and DA config (#436)

## 📝 Summary

New witness calculation algo.
DA config.
Reserve space for builder_tx in block_da_size.

Note:
If da_config.max_da_tx_size is too small, execute function of OpBuilder
will panic, because it's improssible to insert builder transaction.
estimate_builder_tx_da_size may be a little bit heavy, maybe we could
precompute tx size for builder_tx and reuse it

## 💡 Motivation and Context

<!--- (Optional) Why is this change required? What problem does it
solve? Remove this section if not applicable. -->

---

## ✅ I have completed the following steps:

* [x] Run `make lint`
* [x] Run `make test`
* [ ] Added tests (if applicable)

dvush committed:
eth-sparse-mpt: verify proof consistency

dvush committed:
prohibit standard reth root hash impl

dvush committed:
update benchmark

dvush reviewed (approved):
After adding this fix https://github.com/flashbots/rbuilder/pull/429/commits/69ee27be67c493887acb6151e58987613d0fefea I don't see errors anymore on staging, it also does not affect performance since this case only happens when reth syncs new block and we should cancel job anyway.

Since default root hash does not work correctly either (since the problem is in ConsistentDbView I've added this temporarily so we can merge this PR) https://github.com/flashbots/rbuilder/pull/429/commits/29cc0fca432d422241f4ff75fcbcb371e40e6bd1

dvush committed:
grammar fix

ryanschneider commented on crates/rbuilder/src/live_builder/base_config.rs:
```diff
@@ -297,6 +298,10 @@ impl BaseConfig {
         }
+        // temporary guard until reth is fixed
+        if !self.root_hash_use_sparse_trie || self.root_hash_compare_sparse_trie {
+            eyre::bail!("root_hash_use_sparse_trie=true and root_hash_compare_sparse_trie=false must be set, otherwise node will produce incorrect blocks or confusing error messages. These settings are enforced temporarily because upstream parallel root hash implementation is not correct.")
```
@sukoneck we'll need to make sure that we set:

```
root_hash_use_sparse_trie = true
root_hash_compare_sparse_trie=false
```

On the testnet rbuilders once this is merged and then probably revert once fixed.

dvush replied:
This is our production configuration. I don't think we run even one builder without sparse trie. Its mostly for some other setups that may not use it for some reason.

ryanschneider reviewed (approved):
lgtm!  Two potential follow-ups, we will do these as separate PRs:

- Fix `reth-rbuilder` handling of conflicting arguments: https://github.com/flashbots/rbuilder/pull/434/commits/bb52c8f4a8035bcc2f1479c0bff06b0faad193db
- Determine if we still need this blob gas calc change: https://github.com/rkrasiuk/rbuilder/commit/d7b1637967dffc1f4f3ade5303aacc185963c5c3

ferranbt committed:
Fix Op integration tests

ferranbt committed:
Improve logs in rbuilder integration
